gdnative-core/src/core_types/geom/transform.rs

```diff
@@ -157,6 +186,123 @@ impl Transform {
         }
     }
 
+    /// In-place scales basis and origin of the transform by the given scale 
+    /// factor, using matrix multiplication.
+    #[inline]
+    pub fn scale(&mut self, scale:Vector3) {
+        self.scale_basis(scale);
+        self.origin *= scale;
+    }
+
+    /// Scales basis and origin of the transform by the given scale factor, 
+    /// using matrix multiplication.
+    #[inline]
+    pub fn scaled(&self, scale:Vector3) -> Self {
+        let mut copy = *self;
+        copy.scale(scale);
+        copy
+    }
+
+    /// In-place scales basis of the transform by the given scale factor
+    #[inline]
+    fn scale_basis(&mut self, scale:Vector3) {
+        self.basis.scale_self(scale)
+    }
+
+    /// In-place translates the transform by the given offset, relative to 
+    /// the transform's basis vectors.
+    #[inline]
+    pub fn translate_float(&mut self, tx:f32, ty:f32, tz:f32) {
+        self.translate(Vector3::new(tx, ty, tz))
+    }
+
+    /// In-place translates the transform by the given offset, relative to 
+    /// the transform's basis vectors.
+    #[inline]
+    pub fn translate(&mut self, translation:Vector3) {
+        // Note: Godot source uses origin + basis dot translation, 
+        // but self.translate() uses only origin + translation
+        self.origin.x += self.basis.elements[0].dot(translation);
+        self.origin.y += self.basis.elements[1].dot(translation);
+        self.origin.z += self.basis.elements[2].dot(translation);
+    }
+
+    /// Translates the transform by the given offset, relative to 
+    /// the transform's basis vectors.
+    #[inline]
+    pub fn translated_withbasis(&self, translation:Vector3) -> Self {
+        let mut copy = *self;
+        copy.translate(translation);
+        copy
+    }
```

Comment:
This could probably just be named `translated()`.

(same above: `rotated()`, `scaled()`)

Reply:
Yeah I just checked, and if I use previously implemented `Transpose.rs::translated()` it does not match Godot's output of `Transpose.translated()`. Should I remove the old implementation and rename this into `translated()`?